integration_tests/src/main/python/join_test.py

```diff
@@ -1151,3 +1151,63 @@ def do_join(spark):
         conf={"spark.sql.autoBroadcastJoinThreshold": "-1",
               "spark.sql.sources.useV1SourceList": "",
               "spark.rapids.sql.input." + scan_name: False})
+
+@ignore_order(local=True)
+@pytest.mark.parametrize("is_left_host_shuffle", [False, True], ids=idfn)
+@pytest.mark.parametrize("is_right_host_shuffle", [False, True], ids=idfn)
+@pytest.mark.parametrize("is_left_smaller", [False, True], ids=idfn)
+@pytest.mark.parametrize("batch_size", ["1024", "1g"], ids=idfn)
+def test_new_inner_join(is_left_host_shuffle, is_right_host_shuffle, is_left_smaller, batch_size):
+    join_conf = {
+        "spark.rapids.sql.join.useShuffledSymmetricHashJoin": "true",
+        "spark.sql.autoBroadcastJoinThreshold": "1",
+        "spark.rapids.sql.batchSizeBytes": batch_size
+    }
+    left_size, right_size = (2048, 1024) if is_left_smaller else (1024, 2048)
+    def do_join(spark):
+        left_df = gen_df(spark, [
+            ("key1", RepeatSeqGen([1, 2, 3, 4, None], data_type=IntegerType())),
+            ("ints", int_gen),
+            ("key2", RepeatSeqGen([5, 6, 7, None], data_type=LongType())),
+            ("floats", float_gen)], left_size)
+        right_df = gen_df(spark, [
+            ("doubles", double_gen),
+            ("key2", RepeatSeqGen([5, 7, None, 8], data_type=LongType())),
+            ("shorts", short_gen),
+            ("key1", RepeatSeqGen([1, 2, 3, 5, 7, None], data_type=IntegerType()))], right_size)
+        if not is_left_host_shuffle:
```

Comment:
For this I think we are performing a GPU operation to the side that is marked as "not host" so that you can test the "build/stream" side on the "cpu/gpu" set of combinations. It took me a little while to realize that's what this was. If I am right, maybe worth a comment? Just a nit.

Reply:
Yes, the aggregation is applied to get the input to the join to be from a GPU operation rather than a shuffle.  I'll add a comment.